piet/src/samples/mod.rs

```diff
@@ -76,53 +77,67 @@ struct Args {
 /// PNG to the path.
 ///
 /// The `prefix` argument is used for the file names of failure cases.
-pub fn samples_main(f: fn(usize, &Path) -> Result<(), BoxErr>, prefix: &str) -> Result<(), BoxErr> {
-    let args = match Args::from_env() {
-        Ok(args) => args,
-        Err(e) => {
+pub fn samples_main(f: fn(usize, &Path) -> Result<(), BoxErr>, prefix: &str) -> ! {
+    let inner = move || -> Result<(), BoxErr> {
+        let args = Args::from_env()?;
+
+        if args.help {
+            eprintln!("Piet Sample Image Generator\n");
```

Comment:
So I removed it from the help text because that way because printing it when something went wrong felt inapropriate. It read like 1) here is an error, 2) the name of the application is X, 3) here is how to use the app. It felt like 2) interrupted the flow. But obviously it makes sense to write it when someone asks for help.